File: 백준/Silver/11053. 가장 긴 증가하는 부분 수열/가장 긴 증가하는 부분 수열.py
N = int(input())
arr = list(map(int, input().split()))

dp = [1 for _ in range(N)]

for i in range(1, N):
    for j in range(i): # arr의 처음부터 i-1번째 인덱스까지
        if arr[i] > arr[j]: # 숫자의 크기를 비교하여 현재 값이 더 크면
            dp[i] = max(dp[i], dp[j] + 1) # dp배열의 값을 더 큰 값으로 갱신

print(max(dp))

File: 백준/Silver/11053. 가장 긴 증가하는 부분 수열/가장 긴 증가하는 부분 수열.js
const fs = require('fs');

const [N, input] = fs.readFileSync('/dev/stdin').toString().trim().split('\n');

const arr = input.split(' ').map(Number);

const dp = Array(parseInt(N)).fill(1);

for (let i = 1; i < parseInt(N); i++) {
  for (let j = 0; j < i; j++) {
    if (arr[i] > arr[j]) {
      dp[i] = Math.max(dp[i], dp[j] + 1);
    }
  }
}

console.log(Math.max(...dp));